src/main/java/hudson/remoting/Util.java

```diff
@@ -131,4 +136,37 @@ static URLConnection openURLConnection(URL url) throws IOException {
         }
         return con;
     }
+    
+    static InetSocketAddress getResolvedHttpProxyAddress(String host, int port) throws IOException {
+    	InetSocketAddress targetAddress = null;
+    	Iterator<Proxy> proxies = ProxySelector.getDefault().select(URI.create(String.format("http://%s:%d", host, port))).iterator();
+    	// No break -> The while condition already checks if targetAddress is null.
+    	while (targetAddress == null && proxies.hasNext()) {
+    		Proxy proxy = proxies.next();
+    		if (proxy.type() == Proxy.Type.HTTP) {
+    			if (!(proxy.address() instanceof InetSocketAddress))
+    				throw new IOException("Unsupported proxy address type " + proxy.address().getClass());
+    			InetSocketAddress proxyAddress = (InetSocketAddress) proxy.address();
+    			if(proxyAddress.isUnresolved())
+    				proxyAddress = new InetSocketAddress(proxyAddress.getHostName(), proxyAddress.getPort());
+    			targetAddress = proxyAddress;
+    		}
+    	}
+    	if (targetAddress == null) {
+    		String httpProxy = System.getenv("http_proxy");
+    		if (httpProxy != null) {
+    			try {
+    				URL url = new URL(httpProxy);
+    				targetAddress = new InetSocketAddress(url.getHost(), url.getPort());
+    			} catch (MalformedURLException e) {
+    				LOGGER.log(Level.WARNING, INVALID_HTTP_PROXY_MSG + " HTTP Proxy: " + httpProxy);
```

Comment:
Other calls from remoting or Jenkins slave may try to use this corrupted proxy setting. Probably it's safer to just throw <code>IOException</code> and fail here.